core/psh/sysexec.c

```diff
@@ -25,15 +27,90 @@ void psh_sysexecinfo(void)
 }
 
 
+static int psh_sysexec_argmatch(const char *cmd, size_t cmdlen, char const **argv, int argc)
+{
+	int ii;
+	size_t len;
+	const char *currc = cmd, *nextc, *cmdend = cmd + cmdlen - 1;
+
+	for (ii = 0; ii < argc; ii++) {
+
+		if ((nextc = strchr(currc, ' ')) == NULL || nextc > cmdend)
+			nextc = cmdend;
+		len = nextc - currc;
+		if (strlen(argv[ii]) != len || memcmp(argv[ii], currc, len) != 0)
+			return -1;
+
+		if (nextc == cmdend || ii == (argc - 1))
+			break;
+
+		currc = nextc + 1;
+	}
+
+	if (nextc == cmdend && ii == (argc - 1))
+		return 1;
+	else
+		return -1;
+}
+
+
+static int psh_sysexec_checkcommand(int argc, const char **argv)
+{
+	int acc = -1;
+	static const char whitelist[] = PSH_SYSEXECWL;
+	const char *curr, *cmdend;
+	char cmdbuff[80] = "";
+	FILE *wlfile;
+
+	/* check against /etc/whitelist file */
+	if ((wlfile = fopen("/etc/whitelist", "r")) != NULL) {
+		acc = 0;
+		while (fgets(cmdbuff, sizeof(cmdbuff), wlfile) != NULL && acc != 1) {
+			if (isprint(cmdbuff[sizeof(cmdbuff) - 2])) {
+				while (isprint(fgetc(wlfile)))
+					;
+				memset(cmdbuff, 0, sizeof(cmdbuff));
+				continue;
+			}
+
+			cmdend = cmdbuff + strcspn(cmdbuff, "\n");
+
+			if (psh_sysexec_argmatch(cmdbuff, cmdend - cmdbuff + 1, argv, argc) > 0)
```

Comment:
strange way of computing len, could be more readable with straightforward `cmdlen`

Reply:
Changed logic to calculate cmdlen. Thanks